### ISSUE TITLE
CollectionObservable push and unshift not working.

### ISSUE THREAD

**Issue by MrOpperman** [kmalakoff/knockback#95]
I just rolled onto 0.18.1 for testing, and it pretty much broke my code. I was using an unshift on the collectionObservable: 

```
viewModel.demo.collection().unshift()
```

I also tried doing it with push() and they both return undefined. I'm not sure if that is the right way to do it but I was able to in 0.17.2
My declaration looks like this: 

```
demo: kb.collectionObservable(new Demo())
```

my models look like this:

```
var DemoModel = Backbone.Model.extend({
    defaults:{
        demo0: '',
        demo1: '',
        demo2: '',
        demo3: '',
        demo4: '',
        demo5: false
    },
});

var Demo = Backbone.Collection.extend({
    model: DemoModel
});

```

Let me know if you need any more information - I used demo in all these cases because I am using sensitive data.

Note: I can however unshift using this method -  viewModel.demo.collection().unshift(new DemoModel);

So in summary I've realised that the CollectionObservable is no longer aware of the model bound to it after the init phase, and by doing array functions on it you have to stipulate what model you want to use instead of automatically using the default model if no arguments are passed.
